pkg/api/mlflow/dao/repositories/metric.go

```diff
@@ -85,8 +85,19 @@ func (r MetricRepository) CreateBatch(
 		lastIters[lastMetric.Key] = lastMetric.LastIter
 	}
 
+	uniqueContexts := make([]*models.Context, 0, len(metrics))
+	contextMap := make(map[string]*models.Context)
 	latestMetrics := make(map[string]models.LatestMetric)
 	for n, metric := range metrics {
+		if metric.Context != nil {
+			hash := getJsonHash(metric.Context.Json)
+			if contextMap[hash] == nil {
+				contextMap[hash] = metric.Context
+				uniqueContexts = append(uniqueContexts, metric.Context)
+			} else {
```

Comment:
maybe this isn't an else but rather all cases here?

Reply:
actually if the context is not found in the map it would be set to nil and if the context was not found in the map metrics[n].Context already contains the right reference

Reply:
if not found in the map it will be the next time you check (as in the else clause)